### PR TITLE
Implement redirect route helper

### PR DESCRIPTION
**Pull request by alexdavid**
Adds a `redirect` helper to the routes file.

@kevgo @jondistad @charlierudolph Please let me know what you think

### CODE REVIEW

**Comment by charlierudolph** [comment]
Gahh thats weird syntax. I don't think I've ever used that in Rails. Used redirect in the controllers but I don't think ever in the routes file. Thus not sure if this feature is really worth having.

Also the rails guide says redirect in the routes file is a 301 Move Permanently instead of a 302 Found (which apparently should really be 303 See Other)

**Comment by alexdavid** [comment]
Yeah maybe we don't need this. We are using redirects in the routes file of cumulus but those could be moved into controllers I suppose.

**Comment by catharinejm** [comment]
I dunno, I think it could be useful. Probably not often needed, but I have used redirects in rails routes before. It's nice if path structures change, but no computation is required to figure out where to go.

I would leave it in, but may I suggest `GET '/users/:userId', to: '/members/:userId'` instead of 
`%{userId}`?

**Comment by alexdavid** [comment]
I originally went with the `%{}` syntax because that's what rails does. But I agree that `:` would be a better syntax and the goal of this project isn't to follow rails' api, just use it as a guideline.

File: README.md [hunk 1]
```diff
@@ -120,6 +120,19 @@ module.exports = ({DELETE, GET, POST, PUT}) ->
 ```
 
 
+## Redirection
+You can redirect any path to another path using the `redirect` helper
+```coffeescript
+module.exports = ({GET, redirect}) ->
+  GET '/users',  to: redirect('/members')
```
Comment (kevgo) [inline README.md:127]: What about this format?

``` coffeescript
GET '/users',  redirect: '/members'
```

File: test/feature/redirection.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Feature: reverse routing
+
+  Scenario: standard redirection
+    Given the routing definition
+      """
+      GET '/some/route', to: redirect('/redirected/route')
+      """
+    And an exprestive app using defaults
```
Comment (kevgo) [inline test/feature/redirection.feature:8]: I think this line should either move to a `Background` block, or it should be implicit.

File: test/feature/redirection.feature [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    When making a GET request to "/some/route"
+    Then exprestive redirects my request to "/redirected/route"
```
Comment (kevgo) [inline test/feature/redirection.feature]: Then expressive redirects me to "/redirected/route"

Comment (kevgo) [inline test/feature/redirection.feature:10]:
``` cucumber
Then GET requests to "/some/route" are redirected to "/redirected/route"
```

File: test/feature/redirection.feature [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+
+  Scenario: dynamic segment redirection
+    Given the routing definition
+      """
+      GET '/users/:userId/edit', to: redirect('/edit_user/%{userId}')
+      """
+    And an exprestive app using defaults
+    When making a GET request to "/users/1/edit"
```
Comment (kevgo) [inline test/feature/redirection.feature:19]:
``` cucumber
Then GET requests to "/users/1/edit" are redirected to "/edit_user/1"
 And GET requests to "/users/2/edit" are redirected to "/edit_user/2"
```

**Comment by kevgo** [comment]
I think with the syntax simplification that I proposed this could be a useful feature. Example is legacy URL support. Would be nice to have that all declaratively in the routes file, rather than imperatively in the controllers. This way we also don't need controllers for legacy views.

File: README.md (hunk 1 above)
Comment (charlierudolph) [inline README.md:127]: +1

**Comment by alexdavid** [comment]
Going to close this for now since it's stale. Going to re-implement when we need it.

**Pull request closed without merging**